src/rustfmt_diff.rs

```diff
@@ -96,19 +110,27 @@ pub fn print_diff<F>(diff: Vec<Mismatch>, get_section_title: F)
         for line in mismatch.lines {
             match line {
                 DiffLine::Context(ref str) => {
-                    t.reset().unwrap();
+                    if isatty {
+                        t.reset().unwrap();
+                    }
```

Comment:
We should abstract stdout by wrapping it in an enum. We could put the isatty check in the constructor and encapsulate all these `if` expressions in methods.